### ISSUE TITLE
*.scad file not recognize

### ISSUE THREAD

**Issue by Sharif432** [revarbat/openscad_docsgen#18]
% openscad-docsgen -m *.scad
![image](https://user-images.githubusercontent.com/92933919/224881089-15214aee-f5c2-49d7-9b49-9ffc6576cfad.png)

this command doesn't seem to work for me as the error given is "*.scad does not exist". I try out different way like removing the * or just try changing it to "*scad" and it still giving me the same error. My current directory does have .scad file and I even double check using mydir command and it does exist . 

![image](https://user-images.githubusercontent.com/92933919/224880915-d50875a4-2ac0-4cc9-adcd-c43fb813fa73.png)

**Comment by revarbat** [comment]
I'm guessing here, as I don't have a windows machine, but I suspect that Windows does not do automatic command-line argument glob expansion. I can do something about that, luckily.

**Comment by AndrewTasso** [comment]
@revarbat , I'd like to second this one. I ran into the same issue today. It works great in git bash, but not powershell.

Would you be willing to accept a PR to resolve this issue? 

Any path can be fed through `glob.glob()` even if there is no glob in the path string. If there's no glob, it doesn't expand. So, you can treat every path the same, and it'll only expand those which have globs.

My proposed solution is:
1. Set a default value of `"*.scad"` for `srcfiles` argument (using `const="*.scad"`, when adding the `srcfiles` argument to `parser`).
1. Remove the `if not opts.files:` check and globbing of the path.
   - Though, the `if not opts.files:` check can be preserved for the sake of defensive programming.
1. Feed each user provided path through `glob.glob()`, expanding any globs as they're encountered.
1. Flatten the list 
1. Use the flattened list as an argument for the subsequent file checks and parameter to `docsgen.parse_files()`

A quick example of a list comprehension to do the expansions and flatten

```python
    # Expand any glob patterns, leaving paths untouched if not, then flatten into a list of paths
    expanded_files = [files for src_file in opts.files for files in glob.glob(src_file)]
```

**Comment by revarbat** [comment]
The tricky part here is that you don't want to have both the shell glob expanded, then python glob expand that as well, as that can lead to some unlikely but confusing bugs.  If you only `glob.glob()` expand the arguments on a Windows system, this takes care of most of the problem, but the edge case of running from bash on Windows *does* double glob expand the arguments.  This may be acceptable, however, as filenames with glob expansion patterns will be rare.

**Comment by revarbat** [comment]
Oh, and running `openscad-docsgen` with no args is supposed to revert to getting the files list/pattern from the config file.

**Comment by revarbat** [comment]
This should be fixed in v2.0.32

**Comment by AndrewTasso** [comment]
@revarbat I just confirmed that `v2.0.32` resolves this issue. Thank you for the quick turnaround. :)
